lib/tasks/easydeposit2.rake

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+namespace :easydeposit2 do
+  desc 'A rake task to ingest from WoS search CSV dump will be an interim solution to get ED2 ingesting articles again. call like rake easydeposit2:ingest_csv["/path/to/file"]'
+  task :ingest_csv, [:path] => [:environment] do |t, args|
+    csv_file = args[:path]
+    raise(ArgumentError, 'Path cannot be nil') if args[:path].nil?
+    process_ingest_csv(csv_file)
+  end
+end
+
+def process_ingest_csv(path)
+  datetime_today = Time.now.strftime('%Y%m%d%H%M%S') # "20171021125903"
+  logger = ActiveSupport::Logger.new("#{Rails.root}/log/process-import-csv-#{datetime_today}.log")
+  logger.info "Processing ED2 ingest to works in csv: #{path}"
+  csv = CSV.table(path, converters: nil)
+  docs = create_records(logger, csv)
+  logger.info("Create Records with UIDs of: #{docs.uids.inspect}")
+
+  # Pass Records into WebOfScience::ProcessRecords and execute
+  # This calls execute and creates all the Publications and returns the UIDs.
+  # Creating the Publications kicks off the FetchAuthor job which will fail and dead end. Ignore this
+  uids = []
+  # ProcessRecords execute will create publications that are article type and new to database
+  # For testing purpose, add something like 'uids << "WOS:999999999999"' where uid of a known Publication
+  uids += WebOfScience::ProcessRecords.new(docs).execute
+  uids.flatten.compact
+
+  # find Publication by uid
+  # call add_author_email and add abstract on each publication
+  # call EmailArticleRecruitJob this will kick off the author emailing job
+  uids.each do |uid|
+    logger.info("Start email processing for publication with UID: #{uid}")
+    pub = Publication.by_wos_uid(uid).first
+    logger.error("Cannot find Publication of #{uid}") if pub.nil?
+    (emails, abstract) = find_by_uid_csv(csv, uid)
+    pub.update(abstract: abstract) unless abstract.blank?
+    authors = emails.map { |e| { email: e } }
+    pub.add_author_emails(authors)
+    pub.recruit_authors!
```

Comment:
Instead of adding the `bypass_fetch_recruiting_authors` state we can fake the events needed to get this publication to move forward
```
event = Event.create(Event::FETCH_AUTHORS_DIRECTORY_API)

event.update(
  publication: pub,
  message: "Initializing stubbed event for CSV ingest at #{Time.now}",
  restartable: false,
  status: Event::STARTED[:name]
)
event.completed(message: "Completing stubbed event for CSV ingest at #{Time.now}", restartable: false)
```

Reply:
Also the fetch wos event:
```
event = Event.create(Event::FETCH_WOS_CONTENT)

event.update(
  publication: pub,
  message: "Initializing stubbed event for CSV ingest at #{Time.now}",
  restartable: false,
  status: Event::STARTED[:name]
)
event.completed(message: "Completing stubbed event for CSV ingest at #{Time.now}", restartable: false)
```